### PR TITLE
Supports rollback when allocate callback function fails 

### PR DESCRIPTION
**Pull request by wangyang0616**
relate issue [#3777 ](https://github.com/volcano-sh/volcano/issues/3777)

### CODE REVIEW

File: pkg/scheduler/framework/statement.go [hunk 1]
```diff
@@ -225,20 +248,21 @@ func (s *Statement) UnPipeline(task *api.TaskInfo) error {
...
 	}
 
-	hostname := nodeInfo.Name
 	if err := s.ssn.cache.AllocateVolumes(task, hostname, podVolumes); err != nil {
-		return err
+		klog.Errorf("Failed to allocate volume for task %v/%v on node %v when allocating in Session <%v>: %v",
+			task.Namespace, task.Name, hostname, s.ssn.UID, err)
+		errInfos = append(errInfos, err)
 	}
-	defer func() {
```
Comment (Monokaix) [inline pkg/scheduler/framework/statement.go:237]: Can this be deleted directly? Please make sure these rollback  can be done by framework. @JesseStutler

Comment (JesseStutler) [reply pkg/scheduler/framework/statement.go:237]: Yes, framework does in `unallocate`:
https://github.com/volcano-sh/volcano/blob/549ad9ac5aff96343bc5067fa9f736c257d5746b/pkg/scheduler/framework/statement.go#L316-L317

**Pull request merged**